tls/s2n_connection.c

```diff
@@ -326,7 +326,6 @@ int s2n_set_server_name(struct s2n_connection *conn, const char *server_name)
     int len = strlen(server_name);
     if (len > 255) {
         S2N_ERROR(S2N_ERR_SERVER_NAME_TOO_LONG);
-        return -1;
```

Comment:
I looked into -Wunreachable-code as a way to catch this sort of thing. Appears to have been disabled for a while https://gcc.gnu.org/ml/gcc-help/2011-05/msg00360.html .